### PR TITLE
Remove unnecessary dependency, unused configuration and MACROs in ALC5658 driver

### PR DESCRIPTION
**Pull request by btheosam**
ALC5658 driver dont use Worker thread hence stack size
and MSG priority aren't used. ALC5658 buffer's parameters dependency on CONFIG_AUDIO_DRIVER_SPECIFIC_BUFFERS is removed as it isn't required anymore.

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

File: build/configs/artik053/audio/defconfig [hunk 1]
```diff
@@ -421,11 +421,6 @@ CONFIG_AUDIO_DEVICES=y
 # CONFIG_AUDIO_NULL is not set
 CONFIG_AUDIO_ALC5658=y
 CONFIG_ALC5658_INITVOLUME=250
-CONFIG_ALC5658_INFLIGHT=2
-CONFIG_ALC5658_MSG_PRIO=1
-CONFIG_ALC5658_BUFFER_SIZE=2048
```
Comment (Taejun-Kwon) [inline build/configs/artik053/audio/defconfig]: CONFIG_ALC5658_BUFFER_SIZE & NUM_BUFFERS using when CONFIG_AUDIO_DRIVER_SPECIFIC_BUFFERS enabled.
see line962 in alc5658.c

Comment (btheosam) [reply build/configs/artik053/audio/defconfig]: Yes, but now, in this defconfig CONFIG_AUDIO_DRIVER_SPECIFIC_BUFFER is NOT enabled.

Comment (btheosam) [reply build/configs/artik053/audio/defconfig]: Agreed that in latest code, these are used. Hence removed the dependency.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@btheosam You should rewrite commit messages for removing dependancy.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**